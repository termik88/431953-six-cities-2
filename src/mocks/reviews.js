const reviews = [
  {
    id: 1,
    user: {
      id: 15,
      name: `Kendall`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/6.jpg`,
      isPro: false
    },
    rating: 80,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`
  },
  {
    id: 2,
    user: {
      id: 14,
      name: `Corey`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/5.jpg`,
      isPro: true
    },
    rating: 60,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`
  },
  {
    id: 3,
    user: {
      id: 19,
      name: `Christina`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/10.jpg`,
      isPro: false
    },
    rating: 60,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`
  }
];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export default reviews;
