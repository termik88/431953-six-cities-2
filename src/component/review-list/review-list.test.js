import React from 'react';
import renderer from 'react-test-renderer';

import reviews from "../../mocks/reviews.js";
import ReviewList from './review-list.jsx';

describe(`ReviewList component test.`, () => {
  it(`Correctly renders after relaunch`, () => {
    const comments = reviews;
    const mockDate = {
      comments
    };

    const test = renderer
      .create(
          <ReviewList
            {...mockDate}
          />)
      .toJSON();

    expect(test).toMatchSnapshot();
  });
});
