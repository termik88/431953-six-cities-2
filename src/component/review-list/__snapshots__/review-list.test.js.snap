// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList component test. Correctly renders after relaunch 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Review avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/6.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Kendall
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        December 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Review avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/5.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Corey
      </span>
      <span
        className="reviews__user-status"
      >
        Pro
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        December 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Review avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/10.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Christina
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        December 2019
      </time>
    </div>
  </li>
</ul>
`;
